pkg/controller/datavolume/util.go

```diff
@@ -151,6 +160,37 @@ func copyStorageAsPvc(log logr.Logger, storage *cdiv1.StorageSpec) *v1.Persisten
 	return pvcSpec
 }
 
+// Gets either the bound PV or an available satisfying PV
+func getPV(c client.Client, pvcSpec *v1.PersistentVolumeClaimSpec, pvcName, pvcNamespace string) (*corev1.PersistentVolume, error) {
+	if pvcSpec.StorageClassName == nil {
+		return nil, nil
+	}
+
+	pvList := &v1.PersistentVolumeList{}
+	fields := client.MatchingFields{claimStorageClassNameField: *pvcSpec.StorageClassName}
+	if err := c.List(context.TODO(), pvList, fields); err != nil {
+		return nil, err
+	}
+	for _, pv := range pvList.Items {
+		if pv.Status.Phase == corev1.VolumeBound &&
+			pv.Spec.ClaimRef != nil &&
+			pv.Spec.ClaimRef.Name == pvcName &&
+			pv.Spec.ClaimRef.Namespace == pvcNamespace {
```

Comment:
Not sure about this loop.  If a PV exists that meets this criteria, I'm pretty sure it would mean that the target PVC already exists.

I think just the following loop is sufficient

Reply:
fixed